pages/breeds/index.js

```diff
@@ -3,44 +3,118 @@ import Thumbnail from "@/components/Breeds/Thumbnail";
 import Link from "next/link";
 import { styled } from "styled-components";
 
+import { useEffect, useState } from "react";
+import {
+  dogBreedGroups,
+  dogBreedHeight,
+  dogBreedTemperament,
+} from "@/data/dogBreedFilter";
+import { filterDogBreeds } from "@/utils/filterDogBreeds";
+import Wrapper from "@/components/Form/Wrapper.styled";
+import StyledForm from "@/components/Form/Form.styled";
+import StyledList from "@/components/List.styled";
+import StyledLink from "@/components/ListLink.styled";
+import Select from "@/components/Form/Select.styled";
+
+const initialFilter = {
+  group: "all",
+  temperament: "all",
+  size: "all",
+};
+
 export default function BreedList({ dogData }) {
+  const [filter, setFilter] = useState(initialFilter);
+  const [showBreeds, setShowBreeds] = useState(dogData);
+
+  useEffect(() => {
+    filterDogBreeds(filter, dogData, setShowBreeds);
```

Comment:
do you need a useEffect for the filtering?  Normally you can just filter dogData directly and name the result showBreeds (even though I would suggest to name it "filteredBreeds"):
```js
const filteredBreeds = filterDogBreeds(filter,dogData)
```
(instead of using the state you can return the filtered list in your function